### PR TITLE
Convert auth lifecycle to TS, remove dead ILAG code

### PR DESCRIPTION
**Pull request by jryans**
* Converts `Login.js` and `Lifecycle.js` to TS
* Removes various bits of dead ILAG code helpfully revealed by type errors

Part of https://github.com/vector-im/element-web/issues/15350

### CODE REVIEW

File: src/Lifecycle.ts [hunk 1]
```diff
@@ -425,7 +446,10 @@ export function hydrateSession(credentials) {
  *
  * @returns {Promise} promise which resolves to the new MatrixClient once it has been started
  */
-async function _doSetLoggedIn(credentials, clearStorage) {
+async function _doSetLoggedIn(
```
Comment (t3chguy) [inline src/Lifecycle.ts:449]: no need for this to be underscored given it isn't exposed in any way

File: src/Lifecycle.ts [hunk 2]
```diff
@@ -335,7 +356,7 @@ async function _restoreFromLocalStorage(opts) {
     }
 }
 
-async function _handleLoadSessionFailure(e) {
+async function _handleLoadSessionFailure(e: Error): Promise<boolean> {
```
Comment (t3chguy) [inline src/Lifecycle.ts:359]: probably doesn't need underscoring

File: src/Lifecycle.ts [hunk 3]
```diff
@@ -292,8 +313,8 @@ export function getLocalStorageSessionVars() {
 //      The plan is to gradually move the localStorage access done here into
 //      SessionStore to avoid bugs where the view becomes out-of-sync with
 //      localStorage (e.g. isGuest etc.)
-async function _restoreFromLocalStorage(opts) {
-    const ignoreGuest = opts.ignoreGuest;
+async function _restoreFromLocalStorage(opts?: { ignoreGuest?: boolean }): Promise<boolean> {
```
Comment (t3chguy) [inline src/Lifecycle.ts:316]: probably doesn't need underscoring

File: src/Lifecycle.ts [hunk 4]
```diff
@@ -231,7 +239,11 @@ export function handleInvalidStoreError(e) {
     }
 }
 
-function _registerAsGuest(hsUrl, isUrl, defaultDeviceDisplayName) {
+function _registerAsGuest(
```
Comment (t3chguy) [inline src/Lifecycle.ts:242]: probably doesn't need underscoring

**Review by t3chguy — APPROVED** [review]
Looks great otherwise, should we hold this until after the RC just to be triply-safe?

**Comment by jryans** [comment]
> should we hold this until after the RC just to be triply-safe?

Ah yeah, probably a good idea.

**Comment by jryans** [comment]
The CI results no longer match reality, so I'll try just merging... 😱

**Pull request merged**